lib/matplotlib/figure.py

```diff
@@ -2770,11 +2778,62 @@ def clf(self, keep_observers=False):
         self._supxlabel = None
         self._supylabel = None
 
+        if refetch_rcparams:
+            # if refetching rcparams, store their vals in local vars
+            figsize = mpl.rcParams['figure.figsize']
+            dpi = mpl.rcParams['figure.dpi']
+            facecolor = mpl.rcParams['figure.facecolor']
+            edgecolor = mpl.rcParams['figure.edgecolor']
+            frameon = mpl.rcParams['figure.frameon']
+            tight = mpl.rcParams['figure.autolayout']
+            constrained = mpl.rcParams['figure.constrained_layout.use']
+            # then utilize local vars to update Figure attributes...
+            # figsize adjustments
+            if not (
+                np.isfinite(figsize).all()
+                or (np.array(figsize) < 0).any()
+            ):
+                raise ValueError(
+                    f"figure size must be positive finite not {figsize}"
+                )
+            self.bbox_inches = Bbox.from_bounds(0, 0, *figsize)
```

Comment:
OK, this is all copied from `__init__` which is great, except its usually better not to copy (suppose we want to add a new rcParam, now we need to add it in two places).  Can this be cleanly refactored into a helper that is called by both `__init__` and `clf`?  Thanks!